src/components/TextField/TextField.tsx

```diff
@@ -132,21 +132,29 @@ class TextField extends React.PureComponent<CombinedProps, State> {
 
     this.state = {
       height: null,
-      focus: false,
+      focus: props.focused || false,
       id: props.id || getUniqueID(),
     };
   }
 
-  componentDidUpdate({focused}: CombinedProps) {
-    if (
-      this.input &&
-      focused !== this.props.focused &&
-      this.props.focused === true
-    ) {
+  componentDidMount() {
+    if (this.input && this.props.focused) {
       this.input.focus();
     }
   }
 
+  componentDidUpdate({focused: wasFocused}: CombinedProps) {
+    const {focused} = this.props;
+
+    if (this.input) {
+      if (!wasFocused && focused) {
+        this.input.focus();
+      } else if (wasFocused && !focused) {
+        this.input.blur();
```

Comment:
Should we blur when the props change? Using the `focused` prop forces focus on the textfield but omitting the prop doesn't necessarily mean they want the field to be blurred. What are your thoughts on this?

Reply:
In this case, the `blur` even would only trigger if the `focused` prop changed values and it wouldn't be triggered on mount. 

One use case is if `focused` is being set programatically based on an interaction with another component. If the focus is removed (programatically), I'd expect it to trigger the blur event.

I saw this on the [`<SearchField />`](https://github.com/Shopify/polaris-react/tree/master/src/components/TopBar/components/SearchField) component and thought it was a nice addition.

Reply:
Works for me, how about you @solonaarmstrong ?